src/components/chart/SegmentAnalysis.tsx

```diff
@@ -0,0 +1,145 @@
+import * as React from 'react';
+import * as c3 from 'c3';
+import { Typeahead } from "react-bootstrap-typeahead";
+
+interface ComponentProps {
+
+}
+
+interface ComponentState {
+    pushedSegments: [];
+}
+
+// csvData format [[segmentName1, LH1, RH1],[segmentName2, LH2, RH2]]
+const csvData = [
+  ['GSFrontToMargin', 30, 34],
+  ['GOrbital', -20, -16],
+  ['GTemporalMiddle', 10, 14],
+  ['SCentral', 40, 44],
+  ['SFrontSup', -15, -11],
+  ['STemporalInf', 25, 20]
+];
+
+const defaultLeftHemisphere = ['leftHemisphere', 30, -20, 25];
+const defaultRightHemisphere = ['rightHemisphere', 34, -16, 20];
+const defaultChartData = [ defaultLeftHemisphere, defaultRightHemisphere ];
+const defaultSegments = ['GSFrontToMargin', 'GOrbital', 'STemporalInf']
+
+const segments = ['GSFrontToMargin', 'GOrbital', 'GTemporalMiddle',
+                    'SCentral', 'SFrontSup', 'STemporalInf'];
+
+class SegmentAnalysis extends React.Component<ComponentProps, ComponentState> {
+  constructor(props : ComponentProps) {
+    super(props);
+
+    this.state = {
+      pushedSegments: []
+    };
+
+    this.changeData = this.changeData.bind(this);
+  }
+
+  componentDidMount() {
+    this.callChart(defaultChartData, defaultSegments);
+  }
+
+  callChart(inputChart: any, segments: any) {
+    var chart = c3.generate({
```

Comment:
I think we can get rid of this var statement, by adding chart to state instead of having this global var. maybe return the results of c3.generate in the callChart function, and have the caller of this function add the results to the state variable called chart.

Reply:
The var statement is not global. It will just retain the result of the render operation. By default, c3.generate will bind the rendering function to div with id '**chart**'. I have now, exclusively bound the rendering to the component with the corresponding div.

Reply:
@gagan-s-kumar chart is unnecessary here. please remove.

Reply:
var chart is removed